datasource/opennms-kafka/src/main/java/org/opennms/alec/datasource/opennms/OpennmsDatasource.java

```diff
@@ -179,9 +179,12 @@ public void init() throws IOException {
         // Use the class-loader for the KStream class, since the kafka-client bundle
         // does not import the required classes from the kafka-streams bundle
         streams = KafkaUtils.runWithGivenClassLoader(() -> new KafkaStreams(getKTopology(), streamProperties), KStream.class.getClassLoader());
-        if(this.streamStateListener != null) {
-            streams.setStateListener(streamStateListener);
+        
+        if(this.streamStateListener == null) {
+            LOG.debug("Creating a new Kafka Stream Monitor");
+            this.streamStateListener = new KafkaStreamMonitor(this);
```

Comment:
The KafkaStreamMonitor is no longer destroyed when the bundle is stopped/restarted. See reference to `destroy-method` that was removed bellow.